src/DataEditor.tsx

```diff
@@ -1,31 +1,29 @@
-// @flow
-
-import React, { PureComponent } from "react";
-import type { Node } from "react";
+import React from "react";
 import * as Types from "./types";
 import { moveCursorToEnd } from "./util";
 
 type Cell = {
-  value: Node
+  value: Node | string;
 };
 
 type Value = string | number;
 
-export default class DataEditor extends PureComponent<
-  Types.DataEditorProps<Cell, Value>
+class DataEditor extends React.Component<
+  Types.DataEditorProps<Cell, Value>,
+  void
 > {
-  input: ?HTMLInputElement;
+  input?: HTMLInputElement;
 
   static defaultProps = {
     value: ""
   };
 
-  handleChange = (e: SyntheticInputEvent<HTMLInputElement>) => {
+  handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
```suggestion
  handleChange = (e: ChangeEvent<HTMLInputElement>) => {
```
This is a minor opinion but definitely prefer all the used types being imported at the top vs React.* accessed. Make it easy to see what's being used in a given file.